data.sql

```diff
@@ -45,3 +45,23 @@ VALUES ('Blossom',  6,'1998-10-13', 17.0, true, 3, 'Grass-type');
 -- Inserting the seventh animal (Ditto)
 INSERT INTO animals (name, id, date_of_birth, weight_kg, neutered, escape_attempts, species)
 VALUES ('Ditto', 7, '2022-05-14', 22.0, true, 4, 'Normal-type');
+
+--- Task-4 ----
+
+-- Update animals to include species_id based on name
+UPDATE animals
+SET species_id = (CASE
+WHEN name LIKE '%mon' THEN (SELECT id FROM species WHERE name = 'Digimon')
+ELSE (SELECT id FROM species WHERE name = 'Pokemon')
+                 END);
+
+-- Update animals to include owner_id based on name
+UPDATE animals
+SET owner_id = (CASE
+WHEN name IN ('Agumon') THEN (SELECT id FROM owners WHERE full_name = 'Sam Smith')
+WHEN name IN ('Gabumon', 'Pikachu') THEN (SELECT id FROM owners WHERE full_name = 'Jennifer Orwell')
+WHEN name IN ('Devimon', 'Plantmon') THEN (SELECT id FROM owners WHERE full_name = 'Bob')
+WHEN name IN ('Charmander', 'Squirtle', 'Blossom') THEN (SELECT id FROM owners WHERE full_name = 'Melody Pond')
+WHEN name IN ('Angemon', 'Boarmon') THEN (SELECT id FROM owners WHERE full_name = 'Dean Winchester')
+                END);
```

Comment:
- [ ] Kindly add the following data to the `data.sql` file for documentation purposes
```
 INSERT INTO owners (full_name, age)
 VALUES ('Sam Smith', 34);

 INSERT INTO owners (full_name, age)
 VALUES ('Jennifer Orwell', 19);

 INSERT INTO owners (full_name, age)
 VALUES ('Bob', 45);

 INSERT INTO owners (full_name, age)
 VALUES ('Melody Pond', 77);

 INSERT INTO owners (full_name, age)
 VALUES ('Dean Winchester', 14);

 INSERT INTO owners (full_name, age)
 VALUES ('Jodie Whittaker', 38);
```